s/babylon/camera/spectator-camera.ts

```diff
@@ -32,74 +36,48 @@ export function makeSpectatorCamera({
 	const transformB = new TransformNode("camB", scene)
 	const camera = (() => {
 		const name = "spectator camera"
-		const position = new Vector3(0, 5, 0)
+		const position = new Vector3(...startPosition)
 		return new TargetCamera(name, position, scene)
 	})()
 
 	camera.ignoreParentScaling = true
 	camera.parent = transformB
 	transformB.parent = transformA
 
-	let currentLook = v2.zero()
-	function lookAdd(vector: V2) {
-		const radian = Math.PI / 2
-		currentLook = v2.add(currentLook, vector)
-		currentLook[1] = cap(currentLook[1], -radian, radian)
-	}
-
-	function addMouseforce(mouseforce: V2, mouseSensitivity: number) {
-		lookAdd(v2.multiplyBy(mouseforce, mouseSensitivity))
-	}
-
-	function rotateCamera(vectors: V2, mouseSensitivity: number) {
-		addMouseforce(vectors, mouseSensitivity)
-		const [x, y] = currentLook
-		transformB.rotationQuaternion = Quaternion.RotationYawPitchRoll(
-			0, -y, 0,
-		)
-		transformA.rotationQuaternion = Quaternion.RotationYawPitchRoll(
-			x, 0, 0
-		)
-	}
-
-	function moveCam(vector: V2) {
-		const [x, z] = vector
-		const translation = new Vector3(x, 0, z)
-		const newPosition = translation.applyRotationQuaternion(
-			transformB.absoluteRotationQuaternion
-		)
-		transformA.position.addInPlace(newPosition)
-	}
-
-	function applyPointerMovementToCamera([x, y]: V2) {
-		rotateCamera([x, -y], lookSensitivity.mouse)
-	}
+	const state = {currentLook: v2.zero()}
 
 	NubEffectEvent
 		.target(nubContext)
 		.listen(({detail}) => {
 			if (detail.kind === "pointer" && detail.effect === "look") {
 				if (document.pointerLockElement || detail.cause === "Lookpad")
-					applyPointerMovementToCamera(detail.movement)
+					onCameraRotate({
+						getUpdateStarters: () => ({movement: detail.movement}),
+						addMouseforce: (mouseforce) => lookAdd
+							(state, v2.multiplyBy(mouseforce, mouse)),
+						updateCameraRotation: () => transformRotate(state.currentLook, transformA, transformB)
+					})
```

Comment:
this cluster of code is very confusing

following these functions, it feels tangled up -- i think this area and its extracted functions and their dependency functions have become knotted up, and should be rewritten